src/Main.cpp

```diff
@@ -166,8 +169,9 @@ class TCPMessageSender : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
             if (socket) {
                 HiveMindHostSerializer serializer(socket.value());
                 MessageSender messageSender(MessageHandlerContainer::getHostMsgQueue(), serializer,
-                                            m_logger);
+                                            BSPContainer::getBSP(), m_logger);
 
+                messageSender.greet();
```

Comment:
I think the greeting process should be a little more complex than just sending a message. At least, we should wait for a specific response. Otherwise let's say we are connected by USB but on a device which has not yet launched the app. We send a greet and after that, we expect to be connected while we aren't really. I would maybe have a loop that greets at a specific rate (once we now we are physically connected, this will be possible with USB) until we get a confirmation. We should also have a state somewhere saying on which port a successful greeting has been achieved. (Later this could light a LED up).

This is kind of a big change so it could be done in another PR.

Reply:
Would this handshake go in the driver? Because on the TCP side I know that if I am connected, someone is listening on the other side. So the driver would juste send a specific byte while waiting and once it received one, it knows the connection is established like TCP.

Reply:
Well it could go further than just knowing if someone else is on the other end of the line. The handshake could include a version check to be sure messages are compatible between the two entities. I just think it is a good idea to know if we are correctly understood on the other side before trying more messages.

@stlaperriere @g-lesssard What do you guys think?

Reply:
No need for version check though with protobuf, if the fields number id corresponds, they can mismatch and protobuf will only deserialize what it knows

Reply:
But I see what you mean, maybe another PR to add a greetingresponse

Reply:
I agree with the loop mechanism proposed, to make sure we have a connection with the module we want to greet. That is the purpose of a handshake. However, since this is more specific logic, it shouldn't be handled within Propolis but by the HiveMind and HiveConnect firmwares (the HiveMind will greet the ROS-bridge and expect a response and the ESP will greet the HiveMind and expect a repsonse). As you said, a version field could be added in this handshake to detect a breaking change between versions, but this might be more useful in the future than now, as we are currently adding messages more than modifying existing ones.

Reply:
Good, so I will approve and we can create a task to make the handshake loop later on

Reply:
I agree with you guys